packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1635,3 +1674,141 @@ class _FifoCache<K, V> {
     return _cache[key] = loader();
   }
 }
+
+/// Defines the visual density of user interface components.
+///
+/// Density, in the context of a UI, is the vertical and horizontal
+/// "compactness" of the components in the UI. It is unitless, since it means
+/// different things to different UI components.
+///
+/// For example, for buttons, it affects the spacing around the child of the
+/// button. For lists, it affects the distance between baselines of entries in
+/// the list. For chips, it only affects the vertical size, not the horizontal
+/// size.
+class VisualDensity extends Diagnosticable {
```

Comment:
You might mention that the default for the vertical and horizontal density attributes is zero, which means that the component's layout will be just as it appears in the Material Design spec.

Reply:
I also linked in the material design guidance on density.